packages/core/src/rooms.ts

```diff
@@ -0,0 +1,92 @@
+import {Base} from '.'
+
+import {
+  RoomsRequestParameters,
+  RoomsRequestOptions,
+  RoomsResults
+} from './types'
+
+export type RoomsFn = (
+  parameters: RoomsRequestParameters
+) => Promise<RoomsResults>
+
+/**
+ * Creates rooms request string
+ *
+ * @param parameters - rooms request parameters
+ * @param options - rooms request options
+ *
+ * @returns request string
```

Comment:
🐜  I saw we don't follow the official guidelines yet but we started it for core so I think follow them here too. 
```suggestion
 * Creates rooms request string.
 *
 * @param iteneraryParameters - Itenerary request parameters
 * @param options - Rooms request options
 * 
 * @returns request string
 *
 * @example
 * ```
 * const iteneraryParameters = {
 *   hotelId: '12345',
 *   checkIn: '2021-06-16',
 *   checkOut: '2021-06-17',
 *   rooms: '2|2',
 * }
 * const roomRequestOptions = {
 *   deviceType: 'mobile',
 *   locale: 'en',
 *   currency: 'EUR',
 *   anonymousId: 'abcde',
 * }
 *
 * createRoomsRequestString(iteneraryParameters, roomRequestOptions)
 * // hotelId=12345&checkIn='2021-06-16'&...
 * ```
 *
```

Reply:
Please note I renamed the requestParameters to `iteneraryParameters` (could be itineraryRequestParameters if you don't mind the long name?).

For me the distinction between options and parameters wasn't very clear. I would also rename options if I'd have a better name, they seem to be around identifying a user, `userIdentificationParameters` if that's what they are used for?

Reply:
They're coming from different places. Parameters are the search parameters that can be different between different searches and options are the options of a single sapi instance and they cannot be changed, changing of the options means new sapi instance.

Reply:
I propose just to keep them simple just `parameters` and `options` - we have a description in the comments and types. It should be just enough

Reply:
Ah, that's actually how they looked like before your comment 🤣